server/app/views/category.py

```
@@ -0,0 +1,99 @@
+@router.get("/", response_model=PaginatedCategories, operation_id="getCategories")
+async def get_categories(
+    skip: int = Query(0, ge=0, description="Number of records to skip"),
+    limit: int = Query(
+        10, ge=1, le=1000, description="Maximum number of records to return"
+    ),
+    search: str = Query(None, description="Search categories by name"),
+    sort_by: Literal["category_number", "category_name"] = Query(
+        "category_number", description="Field to sort by"
+    ),
+    sort_order: Literal["asc", "desc"] = Query("asc", description="Sort order"),
+    repo: CategoryRepository = Depends(get_category_repository),
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)